### PR TITLE
OCPBUGS-6177: bump(k8s): 1.26.1

### PR DESCRIPTION
**Pull request by sanchezl**
- k8s-1.26: go1.19
- k8s-1.26: bump k8s deps to 1.26.1
- k8s-1.26: go mod vendor

### CODE REVIEW

**Comment by JoelSpeed** [comment]
I have reviewed the diff here, and in particular the changes to existing API types. There's a new field in the `corev1.ResourceRequirements` type which is used in 3 of our APIs. In each case, the content of this field is passed directly to a podspec, so the change is ok to bring in with the new API bump.

This bumps the main API module which is a dependency for others, I will take a look at the `tests` and `tools` submodules as those need not block this merging

**Comment by JoelSpeed** [comment]
The integration tests have found a legitimate issue with the updated CRD definitions, looks like there may be a change in behaviour between 1.25 and 1.26 here. I'm working on getting the integration tests updated for 1.26, once we have those updated, hopefully that will resolve the test failure

**Comment by JoelSpeed** [comment]
Something is wrong with the openapi generation here, looking into it

File: go.mod [hunk 1]
```diff
@@ -5,10 +5,10 @@ go 1.19
 require (
 	github.com/gogo/protobuf v1.3.2
 	golang.org/x/tools v0.1.12
```
Comment (soltysh) [inline go.mod:7]: Nit: would be good to align this with https://github.com/kubernetes/kubernetes/blob/release-1.26/go.mod#L87

Comment (sanchezl) [reply go.mod]: Done.

**Comment by sanchezl** [comment]
Proof PRs:

- https://github.com/openshift/client-go/pull/235
- https://github.com/openshift/library-go/pull/1442
- https://github.com/openshift/apiserver-library-go/pull/95
- https://github.com/openshift/cluster-kube-apiserver-operator/pull/1433
- https://github.com/openshift/cluster-kube-controller-manager-operator/pull/691
- https://github.com/openshift/cluster-kube-scheduler-operator/pull/460

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**